### PR TITLE
ci: enable golangci-lint

### PR DESCRIPTION
**Pull request by aramase**
**Reason for Change**:

- Enable `lint` as part of CI

**Issue Fixed**:

**Notes for Reviewers**:

### CODE REVIEW

File: .pipelines/unit-tests-template.yml [hunk 1]
```diff
@@ -13,6 +13,8 @@ jobs:
           version: 1.15
       - script: make build
         displayName: Build
+      - script: make lint
```
Comment (sozercan) [inline .pipelines/unit-tests-template.yml:16]: should lint be before build so it'll fail fast?

File: .golangci.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+run:
+  timeout: 5m
+  skip-dirs:
+  skip-files:
```
Comment (sozercan) [inline .golangci.yml:4]: nit: let's remove these if it's empty

File: .pipelines/unit-tests-template.yml (hunk 1 above)
Comment (aramase) [reply .pipelines/unit-tests-template.yml:14]: Done!

**Review by sozercan — APPROVED** [review]
LGTM

**Pull request merged**